contracts/modules/tokenizer/TokenizerModule.sol

```diff
@@ -1,38 +1,135 @@
         address ipAssetRegistry,
         address disputeModule
     ) AccessControlled(accessController, ipAssetRegistry) {
+        if (disputeModule == address(0)) revert Errors.TokenizerModule__ZeroDisputeModule();
+
         DISPUTE_MODULE = IDisputeModule(disputeModule);
     }
 
-    function whitelistTokenTemplate(address tokenTemplate, bool allowed) external {}
+    /// @notice Whitelists a token template
+    /// @param tokenTemplate The address of the token template
+    /// @param allowed The whitelisting status
+    function whitelistTokenTemplate(address tokenTemplate, bool allowed) external restricted {
+        if (tokenTemplate == address(0)) revert Errors.TokenizerModule__ZeroTokenTemplate();
+        if (!tokenTemplate.supportsInterface(type(IOwnableERC20).interfaceId))
+            revert Errors.TokenizerModule__UnsupportedERC20(tokenTemplate);
+
+        TokenizerModuleStorage storage $ = _getTokenizerModuleStorage();
+        $.isWhitelistedTokenTemplate[tokenTemplate] = allowed;
+
+        emit TokenTemplateWhitelisted(tokenTemplate, allowed);
+    }
+
+    /// @notice Tokenizes an IP
+    /// @param ipId The address of the IP
+    /// @param tokenTemplate The address of the token template
+    /// @param initData The initialization data for the token
+    /// @return token The address of the newly created token
+    function tokenize(
+        address ipId,
+        address tokenTemplate,
+        bytes calldata initData
+    ) external verifyPermission(ipId) returns (address token) {
+        if (DISPUTE_MODULE.isIpTagged(ipId)) revert Errors.TokenizerModule__DisputedIpId(ipId);
+        if (!IP_ASSET_REGISTRY.isRegistered(ipId)) revert Errors.TokenizerModule__IpNotRegistered(ipId);
+        if (_isExpiredNow(ipId)) revert Errors.TokenizerModule__IpExpired(ipId);
+
+        TokenizerModuleStorage storage $ = _getTokenizerModuleStorage();
+        if (!$.isWhitelistedTokenTemplate[tokenTemplate])
+            revert Errors.TokenizerModule__TokenTemplateNotWhitelisted(tokenTemplate);
+
+        token = address(
+            new BeaconProxy(
+                IOwnableERC20(tokenTemplate).upgradableBeacon(),
+                abi.encodeWithSelector(IOwnableERC20.initialize.selector, initData)
+            )
+        );
 
-    function tokenize(address ipId, address tokenTemplate, bytes calldata initData) external verifyPermission(ipId) {}
+        emit IPTokenized(ipId, token);
+    }
+
+    /// @dev Check if an IP is expired now
+    /// @param ipId The address of the IP
+    function _isExpiredNow(address ipId) internal view returns (bool) {
+        uint256 expireTime = _getExpireTime(ipId);
+        return expireTime != 0 && expireTime < block.timestamp;
+    }
 
+    /// @dev Get the expiration time of an IP
+    /// @param ipId The address of the IP
+    function _getExpireTime(address ipId) internal view returns (uint256) {
+        return IIPAccount(payable(ipId)).getUint256(EXPIRATION_TIME);
+    }
```

Comment:
The function is duplicate with LicenseRegistry. just reuse the function `isExpiredNow` from the LicenseRegistry would be better.

Reply:
The problem is the `_isExpiredNow` is an internal function inside `LicenseRegistry`. Do we want to make it public for external use?

Reply:
There is an external `isExpiredNow` in LicenseRegistry.
https://github.com/storyprotocol/protocol-core-v1/blob/34c5f2b10d3530129c91fb02bdfc4e4befaf354e/contracts/interfaces/registries/ILicenseRegistry.sol#L209

Reply:
I see, thank you for pointing that out. Resolved in [e0d4867](https://github.com/storyprotocol/protocol-periphery-v1/pull/144/commits/e0d486761c0a70d3cd6dc4ef4c39d44cd3b39d92)